### PR TITLE
Add release action

### PR DESCRIPTION
**Pull request by loosebazooka**
Manually triggered release action will tag and generate
java and slsa release artifacts.

Does not sign artifacts with sigstore/gpg yet.

Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

File: .github/workflows/tag-and-build-release.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        required: true
+        default: (for example, 0.1.0)
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check inputs
+        run: |
+          if [[ ! "${{ github.event.inputs.release_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```
Comment (patflynn) [inline .github/workflows/tag-and-build-release.yaml:16]: you don't want to allow SNAPSHOT or ...-BETA, ..-RC1 type version releases?

**Comment by loosebazooka** [comment]
oh goddamnit, looks like I'm spending the rest of the day fixing javadoc errors.

File: gradle.properties [hunk 2]
```diff
@@ -1,2 +1,2 @@
 group=dev.sigstore
-version=0.1-SNAPSHOT
+version=dev-SNAPSHOT
```
Comment (patflynn) [inline gradle.properties:2]: dev?

File: .github/workflows/tag-and-build-release.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
+name: Tag and Build Release
```
Comment (patflynn) [inline .github/workflows/tag-and-build-release.yaml:1]: I used to trigger releases by pushing tags, how does this one get triggered?

File: .github/workflows/tag-and-build-release.yaml (hunk 1 above)
Comment (loosebazooka) [reply .github/workflows/tag-and-build-release.yaml:16]: we could, but I dunno, maybe it doesn't matter right now.

File: gradle.properties (hunk 2 above)
Comment (loosebazooka) [reply gradle.properties:2]: yeah I dunno, I was thinking we avoid the whole dance of constantly updating this version. We already have commits anyway, this number is much less relevant now.

File: .github/workflows/tag-and-build-release.yaml (hunk 3 above)
Comment (loosebazooka) [reply .github/workflows/tag-and-build-release.yaml:1]: You just enter a version number like 0.0.0
This action tags the ref you're on (probably main), then builds it, and publishes the build artifacts to a new release.

It's one giant manually triggered pipeline.

Comment (loosebazooka) [reply .github/workflows/tag-and-build-release.yaml:1]: see: https://github.com/loosebazooka/slsa-java-test/actions/runs/2950354692

**Review by patflynn — DISMISSED** [review]

**Review by patflynn — APPROVED** [review]

**Pull request merged**